base/db/tests/testthat/test.convert_input.R

```diff
@@ -1,10 +1,26 @@
 test_that("`convert_input()` able to call the respective download function for a data item with the correct arguments", {
   mocked_res <- mockery::mock(list(c("A", "B")))
 
-  mockery::stub(convert_input, 'dbfile.input.check', data.frame())
-  mockery::stub(convert_input, 'db.query', data.frame(id = 1))
-  mockery::stub(convert_input, 'PEcAn.remote::remote.execute.R', mocked_res)
-  mockery::stub(convert_input, 'purrr::map_dfr', data.frame(missing = c(FALSE), empty = c(FALSE)))
+  mockery::stub(convert_input, "dbfile.input.check", data.frame())
+  mockery::stub(convert_input, "db.query", data.frame(id = 1))
+  mockery::stub(convert_input, "get.machine.info", list(
+    machine = data.frame(id = 1),
+    input = data.frame(id = 1),
+    dbfile = data.frame(id = 1)
+  ))
+  mockery::stub(convert_input, "PEcAn.remote::remote.execute.R", mocked_res)
+  mockery::stub(convert_input, "check_missing_files", list(
+    result_sizes = data.frame(
+      file = c("A", "B"),
+      file_size = c(100, 200),
+      missing = c(FALSE, FALSE),
+      empty = c(FALSE, FALSE)
+    ),
+    outlist = "test",
+    existing.input = list(data.frame(file = character(0))),
+    existing.dbfile = list(data.frame(file = character(0)))
+  ))
+  mockery::stub(convert_input, "add.database.entries", list(input.id = 1, dbfile.id = 1))
```

Comment:
This feels like a lot of mocks and stubs, but I don't see any that aren't needed. Hopefully as we break up the function further we'll be able to simplify these.

Reply:
Another option would be to pass a custom dataset with known results through the test comparing it with corresponding output for input file. In simple terms, processing data similar to how we wrote test for `met2CF.csv.R`. Is it feasible? Or maybe I am going in the wrong direction?